### DISCUSSION TITLE
Same result for different queries

### DISCUSSION THREAD

**Discussion by Maxttier** [mikro-orm/mikro-orm#6162] [General]
### Describe the bug

```
const user = await this.userService.findUser(id, {
      fields: [
        'id',
      ],
})
```

```
const user = await this.userService.findUser(id, {
      fields: [
        'firstName',
        'id',
        'language',
      ],
})
```

The result is the same `{ id: 1 }` and that is not normal. 
If you do identity map when it need to care about what fields is retrieved. 
Adding refresh or disableIdentityMap to each query.

### Reproduction

-

### What driver are you using?

@mikro-orm/mariadb

### MikroORM version

6

### Node.js version

20

### Operating system

Ubuntu

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by lsh4711** [comment]
https://github.com/lsh4711/reproduction/tree/6153
I tried to reproduce your issue, but it seems to work fine for me.
I need a minimal code example where the problem occurs.
Could you provide a fork of [reproduction](https://github.com/mikro-orm/reproduction)?
